src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -611,7 +611,13 @@ private UpdateLevel GetUpdateLevel(NuGetVersion resolvedVersion, NuGetVersion la
             var sourceRepository = Repository.CreateSource(providers, packageSource, FeedType.Undefined);
```

Comment:
Side note, shouldn't there just be one source repository created per source? It looks like a new one is created for every since ID. Note sure of the perf impact but that seems wrong...

Reply:
I think that's incorrect. 
Just no one caught it.

Reply:
I will leave this out of scope but it does bother me.

Reply:
list package has a bug in it's parallelization too :D 

There are a few things that can be cleaned up at a later point.